src/lexer.coffee

```diff
@@ -514,7 +514,7 @@ exports.Lexer = class Lexer
 
     prev = @prev()
     backslash = prev?[0] is '\\'
-    @seenFor = no unless backslash and @seenFor
+    @seenFor = no unless (backslash or @seenFor?.endsLength < @ends.length) and @seenFor
```

Comment:
There was existing logic to "continue the `for` line" in the case of a backslash line continuer, so just extended that logic to additionally continue the `for` line (ie preserve `@seenFor`) if the linebreak is occurring inside a pair of delimiters ie `{`/`}` or `[`/`]` (as indicated by something having been pushed onto `@ends` since we saw the `FOR` token)